### ISSUE TITLE
BatchNorm feature request: useNonFused to allow 2nd order calculations

### ISSUE THREAD

**Issue by Skylion007** [ppwwyyxx/tensorpack#460]
I came into a rather common use case that would require me to reimplement BatchNorm or modify it since the required functionality does not exist.

Imagine using a GAN loss that requires a 2nd order derivative (DRAGAN or Improved WGAN). You currently cannot use BatchNorm because it usually uses the FusedOP which doesn't have a derivative defined. Either we need to add the Gradient to the Fused BatchNorm or allow users to request to use the non-Fused operation.

**Comment by ppwwyyxx** [comment]
Fused batch norm do have 2nd order gradient IIRC. If not you should report this to tensorflow.
